namelayer-spigot/src/main/java/vg/civcraft/mc/namelayer/database/GroupManagerDao.java

```diff
@@ -100,432 +95,335 @@ public GroupManagerDao(Logger logger, Database db){
-			cleanExecute("drop table moderator;");
-			cleanExecute("alter table faction change `type` group_type varchar(40) not null default 'PRIVATE';");
-			cleanExecute("update faction set group_type = 'PRIVATE';");
-			cleanExecute("alter table faction change founder founder varchar(36);");
-			cleanExecute("alter table db_version add plugin_name varchar(40);");
-			cleanExecute("alter table db_version change db_version db_version int not null;"); // for some reason db_version may have an auto increment on it.
-			cleanExecute("alter table db_version drop primary key;");
-			cleanExecute("insert into permissions (group_id, role, tier) "
+					+ "inner join faction_id fi on fi.group_name = f.group_name;",
+				"drop table moderator;",
+				"alter table faction change `type` group_type varchar(40) not null default 'PRIVATE';",
+				"update faction set group_type = 'PRIVATE';",
+				"alter table faction change founder founder varchar(36);",
+				"insert into permissions (group_id, role, tier) "
 					+ "select f.group_id, 'OWNER', "
 					+ "'DOORS CHESTS BLOCKS OWNER ADMINS MODS MEMBERS PASSWORD SUBGROUP PERMS DELETE MERGE LIST_PERMS TRANSFER CROPS' "
-					+ "from faction_id f;");
-			cleanExecute("insert into permissions (group_id, role, tier) "
+					+ "from faction_id f;",
+				"insert into permissions (group_id, role, tier) "
 					+ "select f.group_id, 'ADMINS', "
 					+ "'DOORS CHESTS BLOCKS MODS MEMBERS PASSWORD LIST_PERMS CROPS' "
-					+ "from faction_id f;");	
-			cleanExecute("insert into permissions (group_id, role, tier) "
+					+ "from faction_id f;",	
+				"insert into permissions (group_id, role, tier) "
 					+ "select f.group_id, 'MODS', "
 					+ "'DOORS CHESTS BLOCKS MEMBERS CROPS' "
-					+ "from faction_id f;");
-			cleanExecute("insert into permissions (group_id, role, tier) "
+					+ "from faction_id f;",
+				"insert into permissions (group_id, role, tier) "
 					+ "select f.group_id, 'MEMBERS', "
 					+ "'DOORS CHESTS' "
 					+ "from faction_id f;");
-			ver = updateVersion(ver, plugin.getName());
-			logger.log(Level.INFO, "Database update to Version one took {0} seconds", (System.currentTimeMillis() - first_time) / 1000);
-		}
-		
-		if (ver == 1){
-			long first_time = System.currentTimeMillis();
-			logger.log(Level.INFO, "Performing database creation!");
-			cleanExecute("create table if not exists faction_id("
+				
+		db.registerMigration(2, false, 
+				new Callable<Boolean>() {
+					@Override
+					public Boolean call() {
+						// Procedures may not be initialized yet.
+						Bukkit.getScheduler().scheduleSyncDelayedTask(NameLayerPlugin.getInstance(), new Runnable(){
+							@Override
+							public void run() {
+								Group g = getGroup(NameLayerPlugin.getSpecialAdminGroup());
+								if (g == null) {
+									createGroup(NameLayerPlugin.getSpecialAdminGroup(), null, null);
+								} else {
+									removeAllMembers(g.getName());
+								}
+							}
+						}, 1);
```

Comment:
You can remove the 1 tick delay parameter. If it's not specified 0 is used which will run the task during the next scheduler run, which for this would be the first tick of the server.  /shrug